core/include/webview/detail/backends/cocoa_webkit.hh

```diff
@@ -650,11 +595,68 @@ private:
     }
     return temp;
   }
+  void m_window_init() {
+    if (!m_owns_window) {
+      return;
+    }
+    objc::autoreleasepool arp;
+    auto setup_delegate_fnc = [this]() {
+      m_window = objc::msg_send<id>("NSWindow"_cls, "alloc"_sel);
+      auto style = NSWindowStyleMaskTitled;
+      m_window = objc::msg_send<id>(
+          m_window, "initWithContentRect:styleMask:backing:defer:"_sel,
+          CGRectMake(0, 0, 0, 0), style, NSBackingStoreBuffered, NO);
+      m_window_delegate = create_window_delegate();
+      objc_setAssociatedObject(m_window_delegate, "webview", (id)this,
+                               OBJC_ASSOCIATION_ASSIGN);
+      objc::msg_send<void>(m_window, "setDelegate:"_sel, m_window_delegate);
+      on_window_created();
+    };
+
+    auto delegate = objc::msg_send<id>(m_app, "delegate"_sel);
+    // Only set the app delegate if it hasn't already been set.
+    if (delegate) {
+      return setup_delegate_fnc();
+    }
+    // See comments related to application lifecycle in create_app_delegate().
+    m_app_delegate = create_app_delegate();
+    objc_setAssociatedObject(m_app_delegate, "webview", (id)this,
+                             OBJC_ASSOCIATION_ASSIGN);
+    objc::msg_send<void>(m_app, "setDelegate:"_sel, m_app_delegate);
+    if (!get_and_set_is_first_instance()) {
+      return setup_delegate_fnc();
+    }
+    // prevent premature run-loop triggering the default sizing
+    set_is_size_set(true);
+
+    // Start the main run loop so that the app delegate gets the
+    // NSApplicationDidFinishLaunchingNotification notification after the run
+    // loop has started in order to perform further initialization.
+    // We need to return from this constructor so this run loop is only
+    // temporary.
+    // Skip the main loop if this isn't the first instance of this class
+    // because the launch event is only sent once. Instead, proceed to
+    // create a window.
+    objc::msg_send<void>(m_app, "run"_sel);
+
+    // dispatch default sizing back onto the event queue
+    if (!m_is_window_shown) {
+      set_is_size_set(false);
+      dispatch_size_default(m_owns_window);
+    }
```

Comment:
Another premature run-loop drain default size gremlin averted!

Reply:
This was actually unneeded because `m_window_init()` all completes before `dispatch_size_default`. It was actually introducing a double dispatch of the default size, so I removed it.